### PR TITLE
ELK Analytics Implementation

### PR DESCRIPTION
**Pull request by heysulo**

### CODE REVIEW

File: components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/ei/EIStatisticsPublisher.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
+/*
```
Comment (heysulo) [inline components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/ei/EIStatisticsPublisher.java:1]: This is file was previously known as `StatisticsPublisher`. No actual changes

File: components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/elasticsearch/schema/ElasticDataSchemaElement.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        try {
+            return (T) attributes.get(key);
+        } catch (ClassCastException e) {
+            return defaultValue;
+        }
+    }
+
+    public String getStartTime() {
+        return Instant.ofEpochMilli(startTime).toString();
```
Comment (arunans23) [inline components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/elasticsearch/schema/ElasticDataSchemaElement.java:48]: What if startTime is not initialized? Shall we handle it?

File: components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/ei/EIStatisticsPublisher.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        if (log.isDebugEnabled()) {
+            ObjectMapper mapper = new ObjectMapper();
+            String jsonString = null;
+            try {
+                jsonString = mapper.writeValueAsString(mapping);
+            } catch (JsonProcessingException e) {
+                log.error("Unable to convert", e);
+            }
+            log.debug("Uncompressed data :");
+            log.debug(jsonString);
```
Comment (arunans23) [inline components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/ei/EIStatisticsPublisher.java:146]: Is there any reason to put it in two lines?

**Comment by heysulo** [comment]
<details>
  <summary>Unit Test Results</summary>
  
  ```python
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for WSO2 Micro Integrator 4.1.0:
[INFO]
[INFO] WSO2 Micro Integrator .............................. SUCCESS [  1.258 s]
[INFO] WSO2 Enterprise Integrator - Component Parent ...... SUCCESS [  0.058 s]
[INFO] WSO2 Carbon - New Data Sources Common .............. SUCCESS [  2.587 s]
[INFO] WSO2 Micro Integrator - Core Component ............. SUCCESS [ 24.629 s]
[INFO] WSO2 Carbon - New Data Sources Core ................ SUCCESS [  8.219 s]
[INFO] WSO2 Carbon - New Data Sources RDBMS ............... SUCCESS [ 11.995 s]
[INFO] WSO2 Carbon - Data source C-app Deployer implementation SUCCESS [  7.876 s]
[INFO] WSO2 Micro Integrator - Server Component ........... SUCCESS [  1.543 s]
[INFO] org.wso2.micro.integrator.security ................. SUCCESS [ 11.117 s]
[INFO] mediation .......................................... SUCCESS [  0.024 s]
[INFO] registry ........................................... SUCCESS [  0.020 s]
[INFO] WSO2 Micro Integrator - Registry ................... SUCCESS [  4.964 s]
[INFO] org.wso2.micro.integrator.coordination ............. SUCCESS [  6.066 s]
[INFO] tasks .............................................. SUCCESS [  0.024 s]
[INFO] WSO2 Carbon - Tasks Core ........................... SUCCESS [  8.154 s]
[INFO] WSO2 Carbon - ntask Integration core ............... SUCCESS [  9.844 s]
[INFO] WSO2 Carbon - Inbound Endpoint Module .............. SUCCESS [  0.690 s]
[INFO] org.wso2.micro.integrator.inbound.endpoint.persistence SUCCESS [  4.685 s]
[INFO] org.wso2.micro.integrator.inbound.endpoint.osgi .... SUCCESS [  4.702 s]
[INFO] org.wso2.micro.integrator.inbound.endpoint ......... SUCCESS [01:53 min]
[INFO] data ............................................... SUCCESS [  0.062 s]
[INFO] data-services ...................................... SUCCESS [  0.019 s]
[INFO] WSO2 Carbon - Data Services Common ................. SUCCESS [  0.836 s]
[INFO] WSO2 Carbon - Data Services SQL Driver ............. SUCCESS [  8.036 s]
[INFO] WSO2 Carbon - Data Services Core ................... SUCCESS [01:10 min]
[INFO] WSO2 Micro Integrator - Mediation Initializer ...... SUCCESS [ 57.869 s]
[INFO] extensions ......................................... SUCCESS [  0.037 s]
[INFO] WSO2 Micro Integrator - Transport Handlers ......... SUCCESS [  5.862 s]
[INFO] org.wso2.micro.integrator.security.handlers ........ SUCCESS [  0.057 s]
[INFO] utsecurity ......................................... SUCCESS [  1.141 s]
[INFO] mediators .......................................... SUCCESS [  0.036 s]
[INFO] WSO2 Carbon - Cache mediator ....................... SUCCESS [  5.856 s]
[INFO] WSO2 Carbon - Identity OAuth Mediator .............. SUCCESS [  5.738 s]
[INFO] WSO2 Micro Integrator - Identity Entitlement Mediator SUCCESS [  6.387 s]
[INFO] WSO2 Micro Integrator - Dataservices mediator ...... SUCCESS [  5.883 s]
[INFO] startup ............................................ SUCCESS [  0.018 s]
[INFO] WSO2 Micro Integrator Mediation StartUp ............ SUCCESS [ 10.116 s]
[INFO] transports ......................................... SUCCESS [  0.050 s]
[INFO] WSO2 Micro Integrator - WebsocketTransport ......... SUCCESS [  5.700 s]
[INFO] data-publishers .................................... SUCCESS [  0.022 s]
[INFO] WSO2 Micro Integrator - Analytics Data Publisher Util SUCCESS [  5.444 s]
[INFO] WSO2 Micro Integrator Analytics - Message Flow Data Publisher SUCCESS [ 11.909 s]
[INFO] WSO2 Micro Integrator - Observablity ............... SUCCESS [  6.870 s]
[INFO] security ........................................... SUCCESS [  0.024 s]
[INFO] WSO2 Carbon - Synapse Security ..................... SUCCESS [  5.432 s]
[INFO] WSO2 Micro Integrator - Extensions ................. SUCCESS [  0.023 s]
[INFO] org.wso2.micro.integrator.management.apis .......... SUCCESS [ 21.064 s]
[INFO] org.wso2.micro.integrator.probes ................... SUCCESS [ 12.269 s]
[INFO] crypto-service ..................................... SUCCESS [  0.017 s]
[INFO] org.wso2.micro.integrator.crypto.impl .............. SUCCESS [  8.543 s]
[INFO] org.wso2.micro.integrator.crypto.provider .......... SUCCESS [  8.971 s]
[INFO] org.wso2.micro.integrator.bootstrap ................ SUCCESS [  4.817 s]
[INFO] WSO2 Carbon - Data service C-app Deployer implementation SUCCESS [  7.277 s]
[INFO] WSO2 Carbon - OData Services Endpoint .............. SUCCESS [  8.396 s]
[INFO] business-adaptors .................................. SUCCESS [  0.030 s]
[INFO] WSO2 Carbon - HL7 Business Messaging - Common ...... SUCCESS [  8.455 s]
[INFO] WSO2 Micro Integrator - Log4j2 Plugins ............. SUCCESS [  6.700 s]
[INFO] WSO2 Carbon - JSR 107 API .......................... SUCCESS [  6.612 s]
[INFO] WSO2 Micro Integrator - Feature Aggregator Module .. SUCCESS [  0.063 s]
[INFO] WSO2 Micro Integrator - Core Server Feature ........ SUCCESS [  4.386 s]
[INFO] WSO2 Micro Integrator - Initializer Server Feature . SUCCESS [  5.096 s]
[INFO] mediation-features ................................. SUCCESS [  0.029 s]
[INFO] security-features .................................. SUCCESS [  0.014 s]
[INFO] WSO2 Micro Integrator Security Feature ............. SUCCESS [  2.196 s]
[INFO] WSO2 Micro Integrator - Mediation NTask Feature .... SUCCESS [  5.360 s]
[INFO] builtin-mediators .................................. SUCCESS [  0.031 s]
[INFO] WSO2 Carbon - All Mediators Server Feature ......... SUCCESS [  0.849 s]
[INFO] inbound-endpoint ................................... SUCCESS [  0.023 s]
[INFO] WSO2 Carbon - Inbound Endpoint Server Feature ...... SUCCESS [  6.607 s]
[INFO] transport-features ................................. SUCCESS [  0.024 s]
[INFO] org.wso2.micro.integrator.websocket.feature ........ SUCCESS [  2.239 s]
[INFO] data-publisher-features ............................ SUCCESS [  0.010 s]
[INFO] WSO2 Micro Integrator - Observablity Feature ....... SUCCESS [  3.848 s]
[INFO] org.wso2.micro.integrator.analytics.messageflow.data.publisher.feature SUCCESS [  4.435 s]
[INFO] WSO2 Micro Integrator - XACML Mediation Feature .... SUCCESS [  2.485 s]
[INFO] WSO2 Micro Integrator - Dataservice Mediation Feature SUCCESS [  1.905 s]
[INFO] data-features ...................................... SUCCESS [  0.033 s]
[INFO] data-services-feature .............................. SUCCESS [  0.021 s]
[INFO] WSO2 Carbon - All Mediators Server Feature ......... SUCCESS [  1.072 s]
[INFO] WSO2 Carbon - Tasks Core Feature ................... SUCCESS [  2.238 s]
[INFO] org.wso2.micro.integrator.crypto.feature ........... SUCCESS [  2.245 s]
[INFO] wso2mi-hl7 ......................................... SUCCESS [  3.924 s]
[INFO] WSO2 Enterprise Integrator - micro-integrator P2 Profile Generation SUCCESS [ 25.874 s]
[INFO] WSO2 Micro Integrator - Integration Tests .......... SUCCESS [  0.077 s]
[INFO] WSO2 MI - Samples .................................. SUCCESS [  3.351 s]
[INFO] WSO2 Micro Integrator - Automation Extensions ...... SUCCESS [  2.182 s]
[INFO] MI Test Common Module .............................. SUCCESS [  3.399 s]
[INFO] MI Integration Test Admin Clients .................. SUCCESS [  3.207 s]
[INFO] MI Integration Test Common Utils ................... SUCCESS [  5.918 s]
[INFO] WSO2 Enterprise Integrator - Mediation Integration Tests SUCCESS [  0.062 s]
[INFO] MI Test Service Module ............................. SUCCESS [  0.435 s]
[INFO] MI test Module-Transport ........................... SUCCESS [40:18 min]
[INFO] MI test Module-Service ............................. SUCCESS [09:43 min]
[INFO] MI test Module-mediator-1 .......................... SUCCESS [15:10 min]
[INFO] MI test Module-mediator-2 .......................... SUCCESS [19:41 min]
[INFO] Mi test Module - Test Patches ...................... SUCCESS [20:25 min]
[INFO] MI test Module - Sample ............................ SUCCESS [03:07 min]
[INFO] MI test Module-Other ............................... SUCCESS [16:47 min]
[INFO] MI Test Module - Platform .......................... SUCCESS [  0.083 s]
[INFO] MI test Module - Platform - RabbitMQ Transport ..... SUCCESS [  5.207 s]
[INFO] MI test Module-Coordination ........................ SUCCESS [  6.975 s]
[INFO] MI test Module - UserStore ......................... SUCCESS [  6.114 s]
[INFO] MI test Module-Transaction ......................... SUCCESS [  4.317 s]
[INFO] MI Integration Test Reporting ...................... SUCCESS [  4.060 s]
[INFO] MI Data Service Tests .............................. SUCCESS [  0.082 s]
[INFO] MI DSS Samples Parent .............................. SUCCESS [  0.017 s]
[INFO] DS Samples Parent .................................. SUCCESS [  0.423 s]
[INFO] DataServices Clients ............................... SUCCESS [ 29.898 s]
[INFO] Mi DSS Integration Test Common ..................... SUCCESS [  0.074 s]
[INFO] MI DSS Admin Clients ............................... SUCCESS [  0.668 s]
[INFO] MI DSS Test Utils .................................. SUCCESS [  0.600 s]
[INFO] WSO2 DataServices Integration Test ................. SUCCESS [  0.049 s]
[INFO] MI DSS Tests ....................................... SUCCESS [01:35 min]
[INFO] WSO2 MI - Management API Tests ..................... SUCCESS [01:56 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:20 h
[INFO] Finished at: 2022-08-05T02:34:11+05:30
[INFO] ------------------------------------------------------------------------
  ```
</details>

**Review by arunans23 — APPROVED** [review]

**Pull request merged**